### ISSUE TITLE
[flake8-django] DJ001 is not raised for models not directly inherited from Model

### ISSUE THREAD

**Issue by konysko** [charliermarsh/ruff#2892]
DJ001 is not raised for models which doesn't directly inherits from django.db.models.Model. The source of bug is in https://github.com/charliermarsh/ruff/blob/f7515739acc0a7b5c580bf05aac6055181739c65/crates/ruff/src/rules/flake8_django/rules/helpers.rs#L6 It should lookup whole mro for the Model class. 

Affected version: 0.0.246
Snippet reproducing bug:
```python
from django.db import models
from django.utils.translation import gettext_lazy as _
from utils.models import BaseModel


class ExampleModel(BaseModel):
    name = models.CharField(max_length=255, verbose_name=_('Name'), null=True)
    description = models.TextField(verbose_name=_('Description'))

    def __str__(self) -> str:
        return self.name
```

**Comment by charliermarsh** [comment]
I think in this case it would also be fine to skip the base model check. If a class has a `django.models.CharField`, that's good enough IMO.
